staff/edit-courses/edit-course.js

```diff
@@ -0,0 +1,359 @@
+
+function getFullText(ele) {
+    let text_html = ele.attributes.onclick.nodeValue;
+    let start = text_html.indexOf("\"");
+    let end = text_html.lastIndexOf("\"");
+    return text_html.substring(start + 1, end);
+}
+
+function waitFilter() {
+    if (timeout) {  
+        clearTimeout(timeout);
+    }
+    timeout = setTimeout(function() {
+        filterTable();
+    }, 150);
+}
+function filterTable() {
+    let input = document.getElementById("filterTableInput");
+    let filter = input.value.toUpperCase();
+    let table = document.getElementById("classTable");
+    let rows = table.getElementsByTagName("tr");
+
+    for (let row of rows)
+    {
+        if (row.cells[0].tagName == "TH") {
+            continue;
+        }
+        for (let column of row.cells) {
+            column.innerHTML = column.innerHTML.replace(HIGHLIGHT[0], ""); // Clean old highlights first
+            column.innerHTML = column.innerHTML.replace(HIGHLIGHT[1], "");
+        }
+        if (filter === "") {
+            row.style.display = "";
+            continue;
+        }
+        let found = false;
+        for (let column of row.cells) {
+            let expand = true;
+            let ele = column.querySelector('.expand');
+            if (ele === null) {
+                ele = column;
+                expand = false;
+            }
+            let posText = ele.innerText.toUpperCase().indexOf(filter);
+            let posHTML = ele.innerHTML.toUpperCase().indexOf(filter);
+            if (posText != -1 && posHTML != -1) {
+                if (expand) {
+                    let full_text = getFullText(ele);
+                    let e = {target: ele};
+                    expandText(e, full_text, true, 1000000, true);
+                    highlightWord(ele, posHTML, posHTML+filter.length);
+                } else {
+                    highlightWord(ele, posHTML, posHTML+filter.length);
+                }
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            row.style.display = "none";
+        } else {
+            row.style.display = "";
+        }
+    }
+    function highlightWord(column, startPos, endPos) {
+        column.innerHTML = column.innerHTML.slice(0, startPos) + HIGHLIGHT[0] + column.innerHTML.slice(startPos, endPos) + HIGHLIGHT[1] + column.innerHTML.slice(endPos);
+    }
+}
+
+function toggleCredits(btn) {
+    timeout = setTimeout(function(btn) {
+        if (btn.ariaPressed === 'true') { // Apparently this isn't set immediately. Timeout works
+            document.getElementById("credits_max_separator").style.display = ""; 
+            document.getElementById("max_hours").style.display = ""; 
+        } else {
+            document.getElementById("credits_max_separator").style.display = "none"; 
+            document.getElementById("max_hours").style.display = "none"; 
+        }
+        btn.value = btn.ariaPressed;
+    }, 1, btn);
```

Comment:
I had no idea you could pass extra arguments to setTimeout. I believe this would also work without arguments if you changed `function(btn) {` to `() => {`.